util/cpuprofile/cpuprofile_test.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cpuprofile
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"runtime/pprof"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/pprof/profile"
+	"github.com/pingcap/tidb/util/testbridge"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+)
+
+func TestMain(m *testing.M) {
+	testbridge.WorkaroundGoCheckFlags()
+	goleak.VerifyTestMain(m)
+}
+
+func TestParallelCPUProfiler(t *testing.T) {
+	if GlobalCPUProfiler != nil {
+		GlobalCPUProfiler.Close()
+	}
+	GlobalCPUProfiler = NewParallelCPUProfiler()
+	GlobalCPUProfiler.Start()
+	defer GlobalCPUProfiler.Close()
+
+	// Test register/unregister nil
+	Register(nil)
+	require.Equal(t, 0, GlobalCPUProfiler.ConsumersCount())
+	Unregister(nil)
+	require.Equal(t, 0, GlobalCPUProfiler.ConsumersCount())
+	require.False(t, GlobalCPUProfiler.inProfilingStatus())
+
+	// Test profile error and duplicate register.
+	dataCh := make(ProfileConsumer, 10)
+	err := pprof.StartCPUProfile(bytes.NewBuffer(nil))
```

Comment:
I'm a bit curious about the purpose of calling `pprof.StartCPUProfile` directly, is that to prevent `GlobalCPUProfiler` from profiling successfully?

Reply:
Yes